src/constructors/full.c

```diff
@@ -131,6 +131,202 @@ igraph_error_t igraph_full(igraph_t *graph, igraph_integer_t n, igraph_bool_t di
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \function igraph_full_multipartite
+ * \brief Create a full multipartite network.
+ *
+ * A multipartite network contains two or more types and connections
+ * are only possible between two vertices in different types.
+ *
+ * \param graph Pointer to an igraph_t object, the graph will be
+ *   created here.
+ * \param types Pointer to an int vector. If not a null pointer,
+ *   it contains information about the vertex types.
+ * \param n Pointer to an int vector, the number of types.
+ * \param directed Boolean, whether to create a directed graph.
+ * \param mode A constant that gives the type of connections for
+ *   directed graphs. If \c IGRAPH_OUT, then edges point from vertices
+ *   of low-index vertices to high-index vertices; if \c
+ *   IGRAPH_IN, then the opposite direction is realized; if \c
+ *   IGRAPH_ALL, then mutual edges will be created.
+ * \return Error code.
+ *
+ * Time complexity: O(|V|+|E|), linear in the number of vertices and
+ * edges.
+ *
+ * \sa \ref igraph_full_bipartite() for full bipartite graphs.
+ */
+igraph_error_t igraph_full_multipartite(igraph_t *graph,
+                          igraph_vector_int_t *types,
+                          const igraph_vector_int_t *n,
+                          igraph_bool_t directed,
+                          igraph_neimode_t mode) {
+    
+    igraph_integer_t no_of_edges = 0;
+    igraph_integer_t no_of_nodes;
+    igraph_vector_int_t edges;
+    igraph_vector_int_t n_acc;
+    igraph_integer_t ptr = 0;
+
+    igraph_integer_t nn = igraph_vector_int_size(n);
+
+    if (nn == 0) {
+        igraph_empty(graph, 0, directed);
```

Comment:
```suggestion
        IGRAPH_CHECK(igraph_empty(graph, 0, directed));
```

All functions that can fail need an `IGRAPH_CHECK`